### PR TITLE
Added support for CSharp8 tests and fixed #2927

### LINKED ISSUE #2927: SA1011 false positive, C#8.0 and NullableContextOptions when used with arrays.
An example:
```
byte[]? data = null; // SA1011 Closing Bracket should be followed by a space
byte[] ? data = null; // SA1018 Nullable type symbol should not be preceded by a space
```

I think SA1011 is a false positive.

### PR DESCRIPTION
**Pull request by vweijsters**
Fixes #2927

### CODE REVIEW

**Comment by vweijsters** [comment]
@sharwell I created this PR to add support for CSharp8. As part of that I had to upgrade Microsoft.CodeAnalysis.CSharp.CodeFix.Testing.XUnit. As a result of this a lot of tests with included documentation fail now. Do you have any pointers on what could have caused that?

**Comment by sharwell** [comment]
I'll take a look

**Comment by sharwell** [comment]
Let's try again after https://github.com/dotnet/roslyn-sdk/pull/307 is merged.

**Comment by vweijsters** [comment]
Works now 👍

File: StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1011ClosingSquareBracketsMustBeSpacedCorrectly.cs [hunk 1]
```diff
@@ -117,7 +117,7 @@ private static void HandleCloseBracketToken(SyntaxTreeAnalysisContext context, S
                     break;
 
                 case SyntaxKind.QuestionToken:
-                    precedesSpecialCharacter = nextToken.Parent.IsKind(SyntaxKind.ConditionalAccessExpression);
+                    precedesSpecialCharacter = nextToken.Parent.IsKind(SyntaxKind.ConditionalAccessExpression) || nextToken.Parent.IsKind(SyntaxKind.NullableType);
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1011ClosingSquareBracketsMustBeSpacedCorrectly.cs]: ❓ Is this needed as part of the initial introduction of the C# 8 harness, or could it be submitted as a follow-up issue to keep this PR clean?

Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1011ClosingSquareBracketsMustBeSpacedCorrectly.cs]: I would have preferred to see #2927 fixed separately due to the size of the rest of the change, but if the CI passes I won't block this PR.

**Comment by sharwell** [comment]
@vweijsters I believe you'll need to update [opencover-report.ps1](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/build/opencover-report.ps1) to run the new test project in AppVeyor.

File: StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1011ClosingSquareBracketsMustBeSpacedCorrectly.cs (hunk 1 above)
Comment (vweijsters) [reply StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1011ClosingSquareBracketsMustBeSpacedCorrectly.cs]: In hindsight that would have been the better option, but given that the fix for #2927 is really small I didn't want to go through the hassle of separating them.

**Pull request merged**